### ISSUE TITLE
Sql-Formatter plugin can not work in IE browser

### ISSUE THREAD

**Issue by xm-c** [sql-formatter-org/sql-formatter#413]
In IE browser，it is report error msg.
![image](https://user-images.githubusercontent.com/57081674/187337857-c54fe7ac-04f4-4767-bc19-3cdcaf1644fe.png)

whether this plugin can support IE, when the version is greater than 2.3.4 ?

**Comment by nene** [comment]
Yes. IE is not supported, as SQL Formatter is built with ES6 target, which isn't supported by IE.

If you need IE support, you have the option to compile it with Babel to ES5.

https://kangax.github.io/compat-table/es6/

**Comment by xm-c** [comment]
@nene I try compile it with Babel to ES5, but I found it also can not work in IE, because The Babel cannot transform unicode regex. 
so. It there any way to compile it?

**Comment by nene** [comment]
What about using [regexpu](https://github.com/mathiasbynens/regexpu) to transform these regexes to ES5?

I suspect though that the main problem is that sql-formatter builds lots of these regexes dynamically, and these can't be detected by these transformation tools.

I think there are really only a few places where we really need these unicode regexes. The main one being [matching of identifiers](https://github.com/sql-formatter-org/sql-formatter/blob/master/src/lexer/regexFactory.ts#L165). And this part definitely can't be transformed, as it's not part of a `/regex-literal/u`.

I think it should be possible to eliminate most of our current unicode regexes (or maybe only use them when unicode support is detected in browser). I'll reopen this issue to reconsider.
